packages/table/src/TableRow.ts

```diff
@@ -80,17 +84,14 @@ export class TableRow extends SpectrumElement {
 
     protected handleClick(event: Event): void {
         if (
-            event
-                .composedPath()
-                .find(
-                    (node) => (node as HTMLElement).localName === 'sp-checkbox'
-                )
+            ['sp-table-cell'].includes((event.target as HTMLElement).localName)
```

Comment:
hmm, things're about to gettings weird here, the reason for this change is for the bug I found: when there is a `selectable component` ( like switch ) on a `selectable` row, clicking on the component may mess up the selection of the row
Video in [the comment](https://github.com/adobe/spectrum-web-components/issues/2807#issuecomment-1421950981)
After refactoring and retest I still see some hiccups as below
- If we only check for `sp-table-cell` like I am suggesting
   - I will solve the issue when clicking on switch on a selectable row, but since it only expects `sp-table-cell` in the logic, when we try to click on the `sp-table-checkbox-cell` ( on the cell itself, not the checkbox ), then the event will be skipped
- If we check for both local names `sp-checkbox` AND `sp-table-checkbox-cell`, the issue still be there, since not only the `sp-checkbox` might be added into the cell, but also others like `sp-color-area` or `sp-switch`